src/analyze/analyze-verify.c

```diff
@@ -283,5 +313,26 @@ int verify_units(char **filenames, UnitFileScope scope, bool check_man, bool run
                         r = k;
         }
 
-        return r;
+        if (set_isempty(s) || r != 0)
+                return r;
+
+        if (IN_SET(recursive_errors, RECURSIVE_ERRORS_YES, RECURSIVE_ERRORS_NO))
+                return -EINVAL;
```

Comment:
EINVAL, isn't there a better way? something more recognizable, to distinguish it from non-syntax errors? maybe ENOTRECOVERABLE